src/Microsoft.AspNetCore.Mvc.ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -209,10 +197,10 @@ public string DisplayTextFor<TResult>(Expression<Func<TModel, TResult>> expressi
                 throw new ArgumentNullException(nameof(expression));
             }
 
-            var modelExplorer = GetModelExplorer(expression);
+            var modelExpression = GetModelExpression(expression);
             return GenerateEditor(
-                modelExplorer,
-                htmlFieldName ?? GetExpressionName(expression),
+                modelExpression.ModelExplorer,
+                htmlFieldName ?? modelExpression.Name,
```

Comment:
This seems to be the only place where there's an additional `GetExpressionName` call where previously there was none. Using `GetModelExpression` seems more consistent than optimizing for the non-null case